### PR TITLE
Add FCH

### PR DESCRIPTION
**Pull request by a-thieme**

### CODE REVIEW

File: std/ndn/fch/fch.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+		return res, e
+	}
+
+	hReq, e := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	if e != nil {
+		return res, e
+	}
+	hReq.Header.Set("Accept", "application/json, text/plain, */*")
+
+	hRes, e := http.DefaultClient.Do(hReq)
```
[the remark this answers is not in the document]
Comment (zjkmxy) [reply std/ndn/fch/fch.go:101]: This comment is accurate. By [document](https://pkg.go.dev/net/http#Client.Do):

> If the returned error is nil, the [Response](https://pkg.go.dev/net/http#Response) will contain a non-nil Body which **the user is expected to close**. If the Body is not **both read to EOF and closed**, the [Client](https://pkg.go.dev/net/http#Client)'s underlying [RoundTripper](https://pkg.go.dev/net/http#RoundTripper) (typically [Transport](https://pkg.go.dev/net/http#Transport)) may not be able to re-use a persistent TCP connection to the server for a subsequent "keep-alive" request.

`io.ReadAll` does not automatically close it.

File: std/ndn/fch/fch.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	"net"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/gorilla/schema"
+)
+
+var encoder = schema.NewEncoder()
```
Comment (zjkmxy) [inline std/ndn/fch/fch.go:21]: Personally, I don't like using a global variable.

File: std/ndn/fch/fch.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	Connect   string `json:"connect"`
+	Prefix    string `json:"prefix,omitempty"`
+}
+
+// Query performs an NDN-FCH query.
+func Query(ctx context.Context, req Request) (res Response, e error) {
+	req.applyDefaults()
+	u, e := req.toURL()
+	if e != nil {
+		return res, e
```
Comment (zjkmxy) [inline std/ndn/fch/fch.go:92]: You don't need to write explicitly the values when using named result parameters.

File: std/ndn/fch/fch.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+	if strings.HasPrefix(hRes.Header.Get("Content-Type"), "application/json") {
+		e = json.Unmarshal(body, &res)
+		return res, e
+	}
+
+	routers := bytes.Split(body, []byte{','})
+	for _, router := range routers {
+		if len(router) == 0 {
+			return res, errors.New("empty response")
```
Comment (zjkmxy) [inline std/ndn/fch/fch.go:122]: Why do not check if `routers` array is empty, but to check individual element instead?

File: std/ndn/fch/fch.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+		}
+
+		connect := string(router)
```
Comment (zjkmxy) [inline std/ndn/fch/fch.go:125]: nit: Connect sounds too much like a function. Maybe rename to `uri`.

**Review by zjkmxy — COMMENTED** [review]
Tested with the following code and it works.

```go
res, err := fch.Query(context.Background(), fch.Request{})
if err != nil {
	println(err)
} else {
	retJson, _ := json.Marshal(res)
	println(string(retJson))
}
```

File: std/ndn/fch/fch.go (hunk 2 above)
Comment (yoursunny) [reply std/ndn/fch/fch.go:21]: gorilla/schema recommends defining `encoder` and `decoder` as package globals.
This allows the instance to cache metadata internally, which improves performance.

File: std/ndn/fch/fch.go (hunk 4 above)
Comment (yoursunny) [reply std/ndn/fch/fch.go:122]: In case the HTTP response body is empty, the `routers` would contain one empty bytearray: https://go.dev/play/p/LYNwIOVRe_B
Hence, it's necessary to check into the list elements.

File: std/ndn/fch/fch.go (hunk 5 above)
Comment (yoursunny) [reply std/ndn/fch/fch.go:125]: It means "connection string" and is the name on the API response.
This is an URI only for WebSocket and WebTransport; in case of UDP, it would be host:port string and not a URI.

File: std/ndn/fch/fch.go (hunk 4 above)
Comment (zjkmxy) [reply std/ndn/fch/fch.go:122]: OK we need document this.

Comment (yoursunny) [reply std/ndn/fch/fch.go:122]: This doesn't need "document".
It needs unit testing that covers the "empty response" case, to ensure this works now and works in the future.

File: std/ndn/fch/fch.go (hunk 3 above)
Comment (a-thieme) [reply std/ndn/fch/fch.go:92]: I don't know what you mean by this

File: std/ndn/fch/fch.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (yoursunny) [reply std/ndn/fch/fch.go:101]: Go standard library does not check for errors from `resp.Body.Close()`.
https://cs.opensource.google/go/go/+/refs/tags/go1.24.4:src/net/http/client.go;l=712

File: std/ndn/fch/fch.go (hunk 3 above)
Comment (a-thieme) [reply std/ndn/fch/fch.go:92]: interesting, thanks

**Comment by a-thieme** [comment]
we'll probably want to squash the commits during the merge to main
